Detector/DetFCCeeCLD/src/GenericCalBarrel_o1_v01_geo.cpp

```diff
@@ -0,0 +1,293 @@
+                slice_pos_z += slice_thickness / 2;
+                // Increment slice number.
+                ++slice_number;
+            }
+            
+#if DD4HEP_VERSION_GE( 0, 15 )
+            //Store "outer" quantities
+            caloLayer.outer_nRadiationLengths = nRadiationLengths;
+            caloLayer.outer_nInteractionLengths = nInteractionLengths;
+            caloLayer.outer_thickness = thickness_sum;
+#endif        
+            
+            // Set region, limitset, and vis.
+            layer_vol.setAttributes(theDetector, x_layer.regionStr(), x_layer.limitsStr(), x_layer.visStr());
+
+
+            
+            // Layer position in Z within the stave.
+            layer_pos_z += layer_thickness / 2;
+            // Layer physical volume.
+            PlacedVolume layer_phv = staveInnerVol.placeVolume(layer_vol, Position(0, 0, layer_pos_z));
+            layer_phv.addPhysVolID("layer", layer_num);
+            layer.setPlacement(layer_phv);
+            
+            //The rest of the data is constant; only the distance needs to be updated  
+            //Store the position up to the inner face of the layer
+            caloLayer.distance = rmin+layer_pos_z + staveThickness / 2 - layer_thickness / 2; 
+            std::cout<<"Layer: "<<layer_num<<" Rmin: "<<rmin<<" layer_pos_z: "<<layer_pos_z<<" Dist: "<<caloLayer.distance
+#if DD4HEP_VERSION_GE( 0, 15 )
+		     <<" inner_thickness: "<<caloLayer.inner_thickness<<" outer_thickness: "<<caloLayer.outer_thickness
+#endif
+		     <<std::endl;
+            //Push back a copy to the caloData structure
+            caloData->layers.push_back( caloLayer ) ;
+            
+            // Increment the layer X dimension.
+            layer_dim_x += layer_thickness * std::tan(layerInnerAngle);
+            // Increment the layer Z position.
+            layer_pos_z += layer_thickness / 2;
+            // Increment the layer number.
+            ++layer_num;
+        }
+        
+        ++layerType;
+    }
+    
+    // Add stave inner physical volume to outer stave volume.
+    staveOuterVol.placeVolume(staveInnerVol);
+    if ( staves )  {
+        // Set the vis attributes of the outer stave section.
+        stave.setVisAttributes(theDetector, staves.visStr(), staveInnerVol);
+        stave.setVisAttributes(theDetector, staves.visStr(), staveOuterVol);
+    }
+    // Place the staves.
+    placeStaves(sdet, stave, rmin, nsides, totalThickness, envelopeVol, innerAngle, staveOuterVol);
+    // Set envelope volume attributes.
+    envelopeVol.setAttributes(theDetector, x_det.regionStr(), x_det.limitsStr(), x_det.visStr());
+    
+    
+    //FOR NOW, USE A MORE "SIMPLE" VERSION OF EXTENSIONS, INCLUDING NECESSARY GEAR PARAMETERS
+    //Copied from Frank's SHcalSc04 Implementation
+    sdet.addExtension< LayeredCalorimeterData >( caloData ) ;
+    
+    return sdet;
+}
+
+DECLARE_DETELEMENT(GenericCalBarrel_o1_v01, GenericCalBarrel_o1_v01_geo )
```

Comment:
Please give them a different name to avoid clashes with the lcgeo library

Reply:
Actually, if only the function name of the creator is changed, please push the detectors to DD4hep/DDDectors instead.

Reply:
@andresailer I'm not sure I follow this comment. you suggest that @EmanuelPerez push this detector to DD4hep, but you will keep it in lcgeo?
I believe we start having too many places with DD4hep code, and we really need to have a discussion on how we want to deal with this for the long term.

Reply:
No, push it to dd4hep, and then we remove it from lcgeo eventually after marking it deprecated. Except that for DD4hep the detector name should be prefixed by `DD4hep_`

Reply:
Hi Andre ! Actually, the lcgeo code needs a minor change anyway, to use a sensitive detector type that is defined in Detector/DetSensitive (in the lcgeo code, the type "calorimeter" is hard-coded). I have changed the name of the GenericCalXX code to CLD_ GenericCalXX, and modified the code such that the sensitive type is taken from the xml, instead of being hard-coded. This requires changes in the xml. I could also push this code into DD4Hep if deemed useful (changing the name to DD4hep_GenericCalXX).